src/easystar.js

```diff
@@ -544,6 +566,19 @@ EasyStar.js = function() {
             return (dx + dy);
         }
     };
+
+    var updateQueueIndex = function () {
+        if (!parallelLimit || !instanceQueue.length) {
+            return;
+        }
+
+        instanceQueueIndex--;
+
+        if (instanceQueueIndex < 0 || instanceQueueIndex > instanceQueue.length) {
```

Comment:
Can you help me understand why this condition would ever be true here?

`instanceQueueIndex > instanceQueue.length`

It seems like we're always decrementing the instanceQueueIndex?

Reply:
i believe you're right. i originally had it increment instead of decrement, so I'll remove that check.